src/HttpClient/Plugin/GzipEncoderPlugin.php

```diff
@@ -16,16 +16,25 @@
  */
 final class GzipEncoderPlugin implements PluginInterface
 {
+    /**
+     * @var StreamFactoryInterface The PSR-17 stream factory
+     */
+    private $streamFactory;
+
     /**
      * Constructor.
      *
+     * @param StreamFactoryInterface $streamFactory The stream factory
+     *
      * @throws \RuntimeException If the zlib extension is not enabled
      */
-    public function __construct()
+    public function __construct(StreamFactoryInterface $streamFactory)
```

Comment:
This is a BC; we could make that argument optional and use the `StreamFactoryDiscovery::find()` paired with a deprecation.

Reply:
You are correct. I've added that, should be all sorted.